src/fees/hooks.ts

```diff
@@ -43,30 +43,29 @@ export function usePaidFees(fees: Fee[]) {
   }
 }
 
-// Returns the maximum amount a user can send, taking into acount gas fees required for the transaction
-// also optionally fetches new fee estimations if the current ones are missing or out of date
 export function useMaxSendAmount(
-  tokenAddress: string | undefined,
+  tokenId: string | undefined,
   feeType: FeeType.SEND | FeeType.SWAP,
   shouldRefresh: boolean = true
 ) {
   const dispatch = useDispatch()
-  const balance = useTokenInfoByAddress(tokenAddress)?.balance ?? new BigNumber(0)
+  const balance = useTokenInfo(tokenId!)?.balance ?? new BigNumber(0)
   const feeEstimates = useSelector(feeEstimatesSelector)
+  const tokenInfo = useTokenInfo(tokenId!)
 
   // Optionally Keep Fees Up to Date
   useEffect(() => {
-    if (!shouldRefresh || !tokenAddress) return
-    const feeEstimate = feeEstimates[tokenAddress]?.[feeType]
+    if (!shouldRefresh || !tokenInfo?.address) return
+    const feeEstimate = feeEstimates[tokenInfo.address]?.[feeType]
     if (
       (feeType === FeeType.SWAP && balance.gt(0)) ||
       !feeEstimate ||
       feeEstimate.error ||
       feeEstimate.lastUpdated < Date.now() - ONE_HOUR_IN_MILLIS
     ) {
-      dispatch(estimateFee({ feeType, tokenAddress }))
+      dispatch(estimateFee({ feeType, tokenAddress: tokenInfo.address }))
     }
-  }, [tokenAddress, shouldRefresh])
+  }, [tokenId, shouldRefresh])
```

Comment:
I had to change this to tokenInfo?.tokenId since tokenInfo is an object was created on render using it in the dependency array causes fee's to be re-estimated ever render.